integration/cypress/integration/legacy/subnets.spec.ts

```diff
@@ -16,5 +17,43 @@ context("Subnets", () => {
       "href",
       generateLegacyURL("/networks?by=fabric")
     );
+    cy.findByRole("navigation", { name: "primary" }).within(() => {
+      cy.findByRole("link", { current: "page" }).should(
+        "have.attr",
+        "href",
+        generateLegacyURL("/networks?by=fabric")
+      );
+    });
+  });
+
+  it("displays the main networking view correctly", () => {
+    const expectedHeaders = [
+      "Fabric",
+      "VLAN",
+      "DHCP",
+      "Subnet",
+      "Available IPs",
+      "Space",
+    ];
+    cy.findByRole("table", { name: "Subnets" }).within(() => {
+      expectedHeaders.forEach((name) => {
+        cy.findByRole("columnheader", { name }).should("exist");
+      });
+    });
+  });
+
+  it("allows filtering by 'fabrics' or 'spaces' via the 'Group by' drop-down", () => {
```

Comment:
It seems like an odd approach to write tests for the legacy version only to have to port them over to the React version soon afterwards. It also sets a precedent of using the legacy app as a goalpost, when we really are trying to replace it wholesale - we use it as a starting point, but we try not to replicate things that are obviously broken or have some awkward UX that can be easily fixed.

It obviously doesn't hurt to include these tests now even if they do get replaced and changed a bit, but I'm curious of the reasoning.

Reply:
I cannot imagine having no end to end coverage for a feature that I'm working on therefore I would write them anyway - but doing this first gives an ability to almost do TDD approach moving forward - writing the new implementation and watching the tests go green as you go.

> It seems like an odd approach to write tests for the legacy version only to have to port them over to the React version soon afterwards.

The way I think about it is - I didn't write tests for the _legacy version_. The goal was to write tests for critical user journeys, which will remain the same after migration or even some redesigns (e.g. link inside of a primary navigation will still be a link inside of primary navigation even after we move to using [application panels](https://vanillaframework.io/docs/layouts/application#panels-example) in the future, which I know is planned).

>  we try not to replicate things that are obviously broken or have some awkward UX that can be easily fixed.

That's also the reason I decided to write tests first - user-centric testing approach is excellent in identifying accessibility issues/HTML that has questionable structure.

> It obviously doesn't hurt to include these tests now even if they do get replaced and changed a bit, but I'm curious of the reasoning.

I didn't want to throw away everything but analyse the current implementation and keep the good stuff (if it ain't broke, don't fix it).

This is a side-effect of me doing preparatory work - going through code and UI, analysing the feature and writing them to serve as a documentation as I go. Just a part of my regular process.